pkg/streamingpromql/engine_test.go

```diff
@@ -1889,3 +1889,30 @@ func TestCompareVariousMixedMetricsRate(t *testing.T) {
 
 	runMixedMetricsTests(t, expressions, pointsPerSeries, seriesData)
 }
+
+func TestCompareVariousMixedMetricsComparisonOps(t *testing.T) {
+	labelsToUse, pointsPerSeries, seriesData := getMixedMetricsForTests()
+
+	// Test each label individually to catch edge cases in with single series
+	labelCombinations := testutils.Combinations(labelsToUse, 1)
+	// Generate combinations of 2 labels. (e.g., "a,b", "e,f", etc)
+	labelCombinations = append(labelCombinations, testutils.Combinations(labelsToUse, 2)...)
+
+	expressions := []string{}
+
+	for _, labels := range labelCombinations {
+		labelRegex := strings.Join(labels, "|")
+		for _, op := range []string{"==", "!=", ">", "<", ">=", "<="} {
```

Comment:
This will only test vector / scalar combinations. Should we also add some cases for vector / vector combinations?

Reply:
Good idea, done.

Are we testing scalar / scalar cases anywhere? (they should't go here, but I only recall a few basic comparisons upstream)

Reply:
>Are we testing scalar / scalar cases anywhere? (they should't go here, but I only recall a few basic comparisons upstream)

Yep, [here](https://github.com/grafana/mimir/blob/45a14c913b0433003b28d17cc1f958dcfe72fd14/pkg/streamingpromql/testdata/ours/binary_operators.test#L703-L720).